pkg/xds/generator/outbound_proxy_generator.go

```diff
@@ -148,6 +148,18 @@ func (o OutboundProxyGenerator) generateCDS(ctx xds_context.Context, proxy *mode
 			if clusters.Get(clusterName).RequireTLS() {
 				edsClusterBuilder = edsClusterBuilder.Configure(envoy_clusters.ClientSideTLS(proxy.OutboundTargets[serviceName][0].Target))
 			}
+			protocol := o.inferProtocol(proxy, []envoy_common.ClusterSubset{
+				{
+					ClusterName: clusterName,
+					Weight:      0,
+					Tags:        clusters.Tags(clusterName)[0],
+				},
+			})
```

Comment:
Can you add `Subsets()` method to `Cluster` and do
```
protocol := o.inferProtocol(proxy, clusters.Get(clusterName).Subsets())
```
?

I'd rather check all subsets, just to be sure.

Can we have a unit test for this? (I'd add this to one of the current test in outbound_proxy_generator_test.go)